app/src/main/java/com/teamounce/ounce/settings/SettingsActivity.kt

```diff
@@ -0,0 +1,135 @@
+                            val intent = Intent (this@SettingsActivity,SettingsActivity::class.java )
+                            startActivity(intent)
+                        }
+                    })
+                    .create()
+                dialog.show(supportFragmentManager, dialog.tag)
+        }
+
+        //탈퇴하기 버튼 눌렀을 때
+        txt_ounceout.setOnClickListener {
+            val dialog = SettingCustomDialogBuilder()
+                .setTitle("탈퇴 시 계정 정보,기록 등\n모든 것이 삭제돼요.\n정말 탈퇴하시겠어요?")
+                .setPositiveButton("네")
+                .setNegativeButton("잘못 눌렀어요")
+                .setButtonClickListener(object : SettingCustomDialogListener{
+                    override fun onClickPositiveButton() {
+                        Toast.makeText(this@SettingsActivity,"ㅋ", Toast.LENGTH_SHORT).show()
+                        val intent = Intent(this@SettingsActivity,LoginActivity::class.java)
+                        startActivity(intent)
+                    }
+
+                    override fun onClickNegativeButton() {
+                        Toast.makeText(this@SettingsActivity,"", Toast.LENGTH_SHORT).show()
+                        val intent = Intent (this@SettingsActivity,SettingsActivity::class.java )
+                        startActivity(intent)
+                    }
+                })
+                .create()
+
+            dialog.show(supportFragmentManager, dialog.tag)
+
+        }
+    }
+
+    fun settingButton(){
+        settingButtonAdapter = SettingButtonAdapter(this)
+        recyclerview_setting_buttons.adapter = settingButtonAdapter
+        recyclerview_setting_buttons.addItemDecoration(VerticalItemDecoration(12))
+        loadSettingButton()
+
+    }
+
+
+    fun loadSettingButton(){
+        datas.apply {
+            add(
+                SettingButtonData(
+                    R.drawable.ic_cat,
+                    "고양이 관리"
+                )
+            )
+            add(
+                SettingButtonData(
+                    R.drawable.ic_notice,
+                    "공지사항"
+                )
+            )
+
+            add(
+                SettingButtonData(
+                    R.drawable.ic_tos,
+                    "이용약관"
+                )
+            )
+
+            add(
+                SettingButtonData(
+                    R.drawable.ic__question,
+                    "문의하기"
+                )
+            )
+
+            add(
+                SettingButtonData(
+                    R.drawable.ic_backup,
+                    "기록 백업 및 복원"
+                )
+            )
+
+            settingButtonAdapter.datas = datas
+            settingButtonAdapter.notifyDataSetChanged()
+        }
+    }
+    fun ounceversion(context:Context){
+        val info:PackageInfo = context.packageManager.getPackageInfo(context.packageName,0)
+        val version = info.versionName
+
+        txt_ounceversion.text = version
+    }
```

Comment:
잘 찾으셨군요! 칭찬 박습니다 우어아아아아아ㅏㅇ


+) context를 굳이 넣을 필요가 없다
```suggestion
    fun ounceversion(){
        val info:PackageInfo = packageManager.getPackageInfo(context.packageName,0)
        val version = info.versionName

        txt_ounceversion.text = version
    }
```